backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskResource.java

```diff
@@ -171,8 +171,11 @@ public Response copy(Action action) {
 
     @Override
     public Response reduce(Action action) {
-        Disk disk = get();
-        Guid imageId = getDiskImageId(disk.getImageId());
+        Guid imageId = Guid.createGuidFromString(action.getDisk().getImageId());
+        if (imageId == null) {
+            Disk disk = get();
+            imageId = getDiskImageId(disk.getImageId());
+        }
         ImagesActionsParametersBase params = new ImagesActionsParametersBase(imageId);
```

Comment:
How do we protect the image when the image if the vm or another storage operations are writing to it?

For running VM, we can have:
- base volume of live merge
- target volume of disk replication

For storage operations:
- base volume in cold merge
- target volume in copy_data
- upload disk volume

This is probably already correct, but since we introduce new use case which was not possible before we need to check.

Reply:
don't these merge operations done only when removing a snapshot? because in that case the disk is locked and the Reduce operation also locks the disk